connect/proxy/proxy_test.go

```diff
@@ -60,12 +61,15 @@ func TestProxy_public(t *testing.T) {
 	defer p.Close()
 	go p.Serve()
 
+	// We create this client with an explicit ServerNextProtos here for safety, so
+	// we can properly verify that h2 was not accepted below
```

Comment:
```suggestion
	// We create this client with an explicit ServerNextProtos here which will use `h2`
	// if the proxy supports it. This is so we can verify below that the proxy _doesn't_
	// advertise `h2` support as it's only a L4 proxy.
```

Reply:
This is a nit pick - see what you think but it took me a second to figure out the mechanics here and this wording might possibly make it a bit more explicit?